### PR TITLE
use null to overwrite when merge jObjects

### PR DESCRIPTION
**Pull request by mingwli**
https://ceapex.visualstudio.com/Engineering/_workitems/edit/81106/

### CODE REVIEW

File: src/docfx/lib/json/JsonUtility.cs [hunk 1]
```diff
@@ -233,7 +233,7 @@ public static (List<Error>, JToken) Parse(string json, string file)
             }
         }
 
-        public static void Merge(JObject container, JObject overwrite)
+        public static void Merge(JObject container, JObject overwrite, bool overwriteWithNull = false)
```
Comment (OsmondJiang) [inline src/docfx/lib/json/JsonUtility.cs]: that's should also apply to OverwriteConfig, would you please help to change and add a small test for that?

Comment (mingwli) [reply src/docfx/lib/json/JsonUtility.cs]: plan to make the `overwrite-with-null` default behavior instead of just a flag

Comment (yufeih) [inline src/docfx/lib/json/JsonUtility.cs]:
```suggestion
        public static void Merge(JObject container, JObject overwrite, bool ignoreNulls = false)
```

File: src/docfx/lib/json/JsonUtility.cs [hunk 2]
```diff
@@ -245,9 +245,9 @@ public static void Merge(JObject container, JObject overwrite)
 
                 if (container[key] is JObject containerObj && value is JObject overwriteObj)
                 {
-                    Merge(containerObj, overwriteObj);
+                    Merge(containerObj, overwriteObj, overwriteWithNull);
                 }
-                else if (IsNullOrUndefined(container[key]) || !IsNullOrUndefined(value))
+                else if (overwriteWithNull || IsNullOrUndefined(container[key]) || !IsNullOrUndefined(value))
```
Comment (yufeih) [inline src/docfx/lib/json/JsonUtility.cs]: Test `ignoreNulls` flag in `TestJsonMerge`

File: src/docfx/lib/json/JsonUtility.cs (hunk 1 above)
Comment (live1206) [reply src/docfx/lib/json/JsonUtility.cs]: `overwriteWithNull` means `NotIgnoreNulls`

File: src/docfx/lib/json/JsonUtility.cs [hunk 3]
```diff
@@ -233,7 +233,7 @@ public static (List<Error>, JToken) Parse(string json, string file)
             }
         }
 
-        public static void Merge(JObject container, JObject overwrite)
+        public static void Merge(JObject container, JObject overwrite, bool overwriteWithNull = true)
```
Comment (yufeih) [inline src/docfx/lib/json/JsonUtility.cs]:
```suggestion
        public static void Merge(JObject container, JObject overwrite, NullValueHandling nullValueHandling = NullValueHandling.Include)
```

File: src/docfx/build/metadata/MetadataProvider.cs [hunk 4]
```diff
@@ -49,7 +49,7 @@ public MetadataProvider(Config config)
             if (yamlHeader != null)
             {
                 errors.AddRange(MetadataValidator.ValidateGlobalMetadata(yamlHeader));
-                JsonUtility.Merge(result, yamlHeader);
+                JsonUtility.Merge(result, yamlHeader, overwriteWithNull: true);
```
Comment (live1206) [inline src/docfx/build/metadata/MetadataProvider.cs]: isn't `overwriteWithNull: true` default value already?

File: src/docfx/config/ConfigLoader.cs [hunk 5]
```diff
@@ -3,18 +3,18 @@
...
 namespace Microsoft.Docs.Build
 {
     internal static class ConfigLoader
     {
-        private const string Files = "files";
-        private const string Exclude = "exclude";
         private const string Extend = "extend";
         private const string DefaultLocale = "defaultLocale";
         private const string Localization = "localization";
+        private static readonly string[] KeysToExpand = new string[] { "files", "exclude", "extend" };
```
Comment (yufeih) [inline src/docfx/config/ConfigLoader.cs]:
```suggestion
        private static readonly string[] s_keysToExpand = new string[] { "files", "exclude", "extend" };
```

Comment (OsmondJiang) [inline src/docfx/config/ConfigLoader.cs]: reuse the const string

File: src/docfx/cli/CommandLineOptions.cs [hunk 6]
```diff
@@ -16,14 +16,20 @@ internal class CommandLineOptions
 
         public JObject ToJObject()
         {
+            var output = new JObject
+            {
+                ["path"] = Output != null ? (JValue)Output : JValue.CreateNull(),
```
Comment (yufeih) [inline src/docfx/cli/CommandLineOptions.cs]:
```csharp
if (Output != null)
    output["path"] = Output,
```

File: src/docfx/config/ConfigLoader.cs [hunk 7]
```diff
@@ -201,16 +201,18 @@ private static void OverwriteConfig(JObject config, string locale, string branch
...
+                    config[key] = ExpandStringArray(config[key]);
+            }
             return config;
         }
 
         private static JArray ExpandStringArray(JToken e)
         {
-            if (e is null)
-                return null;
+            Debug.Assert(!(e is null));
```
Comment (OsmondJiang) [inline src/docfx/config/ConfigLoader.cs]: we can use `e != null`....

**Pull request merged**